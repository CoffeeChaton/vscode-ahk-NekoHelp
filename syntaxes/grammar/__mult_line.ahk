~F12::Reload
~F11::ExitApp

MsgBox,
    ( LTrim ; has LTrim, this line deep === test2 line deep +1
        deep+1...
        deep+1...
        deep+1...
    )

MsgBox, % "
    ( LTrim ; has LTrim, this line deep === test2 line deep +1
        deep+1...
        deep+1...
        deep+1...
    )"

MsgBox, % "
    ( ;not has LTrim, this line deep === test2 line deep +1
deepRaw...
    deepRaw...
            deepRaw...
deepRaw...
                            deepRaw...
                        deepRaw...
    )"

MsgBox,
    ( ;not has LTrim, this line deep === test2 line deep +1
deepRaw...
    deepRaw...
            deepRaw...
deepRaw...
                            deepRaw...
                        deepRaw...
    )


test1 := "
    ( LTrim ; has LTrim, this line deep === test2 line deep +1
        deep+1...
        deep+1...
        deep+1...
    )" ; this line deep === test2 line deep +1
----------------------------------------------------------------------------------------------------
test2 := "
        (  ;not has LTrim, this line deep === test2 line deep +1
deepRaw...
    deepRaw...
            deepRaw...
deepRaw...
                            deepRaw...
                        deepRaw...
        )" ;this line deep === test2 line deep +1


bar(str1, opt) {

}

foo() {

    bar("
        (
            w            -> up
            s            -> down
            a            -> left
            d            -> right
        )", "w",0)
    ;    ^mulit-line end
}
