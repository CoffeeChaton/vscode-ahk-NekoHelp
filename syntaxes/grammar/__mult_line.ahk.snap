>~F12::Reload
#^ source.ahk meta.hotkey.line.ahk storage.modifier.hotkey.ahk
# ^^^ source.ahk meta.hotkey.line.ahk string.2.storage.modifier.hotkey.ahk markup.bold.hotkey.ahk
#    ^^ source.ahk meta.hotkey.line.ahk
#      ^^^^^^ source.ahk keyword.control.flow.ahk
>~F11::ExitApp
#^ source.ahk meta.hotkey.line.ahk storage.modifier.hotkey.ahk
# ^^^ source.ahk meta.hotkey.line.ahk string.2.storage.modifier.hotkey.ahk markup.bold.hotkey.ahk
#    ^^ source.ahk meta.hotkey.line.ahk
#      ^^^^^^^ source.ahk keyword.control.flow.ahk
>
>MsgBox,
#^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#      ^ source.ahk meta.command.line.ahk punctuation.ahk
>    ( LTrim ; has LTrim, this line deep === test2 line deep +1
#^^^^^ source.ahk string.multiline.ahk
#     ^^^^^^ source.ahk string.multiline.ahk keyword.multiline.ahk
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk comment.line.semicolon.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>    )
#^^^^^ source.ahk string.multiline.ahk
>
>MsgBox, % "
#^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#      ^ source.ahk meta.command.line.ahk punctuation.ahk
#       ^ source.ahk meta.command.line.ahk
#        ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#         ^ source.ahk meta.command.line.ahk
#          ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
>    ( LTrim ; has LTrim, this line deep === test2 line deep +1
#^^^^^ source.ahk string.multiline.ahk
#     ^^^^^^ source.ahk string.multiline.ahk keyword.multiline.ahk
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk comment.line.semicolon.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>    )"
#^^^^^^ source.ahk string.multiline.ahk
>
>MsgBox, % "
#^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#      ^ source.ahk meta.command.line.ahk punctuation.ahk
#       ^ source.ahk meta.command.line.ahk
#        ^ source.ahk meta.command.line.ahk punctuation.definition.variable.percent.ahk
#         ^ source.ahk meta.command.line.ahk
#          ^ source.ahk meta.command.line.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
>    ( ;not has LTrim, this line deep === test2 line deep +1
#^^^^^ source.ahk string.multiline.ahk
#     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk comment.line.semicolon.ahk
>deepRaw...
#^^^^^^^^^^^ source.ahk string.multiline.ahk
>    deepRaw...
#^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>            deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>deepRaw...
#^^^^^^^^^^^ source.ahk string.multiline.ahk
>                            deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>                        deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>    )"
#^^^^^^ source.ahk string.multiline.ahk
>
>MsgBox,
#^^^^^^ source.ahk meta.command.line.ahk support.function.ahk
#      ^ source.ahk meta.command.line.ahk punctuation.ahk
>    ( ;not has LTrim, this line deep === test2 line deep +1
#^^^^^ source.ahk string.multiline.ahk
#     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk comment.line.semicolon.ahk
>deepRaw...
#^^^^^^^^^^^ source.ahk string.multiline.ahk
>    deepRaw...
#^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>            deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>deepRaw...
#^^^^^^^^^^^ source.ahk string.multiline.ahk
>                            deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>                        deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>    )
#^^^^^ source.ahk string.multiline.ahk
>
>
>test1 := "
#^^^^^^ source.ahk
#      ^^ source.ahk keyword.operator.assignment.ahk
#        ^ source.ahk
#         ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
>    ( LTrim ; has LTrim, this line deep === test2 line deep +1
#^^^^^ source.ahk string.multiline.ahk
#     ^^^^^^ source.ahk string.multiline.ahk keyword.multiline.ahk
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk comment.line.semicolon.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        deep+1...
#^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>    )" ; this line deep === test2 line deep +1
#^^^^^^ source.ahk string.multiline.ahk
#      ^ source.ahk meta.comment.line.semicolon.ahk
#       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>----------------------------------------------------------------------------------------------------
#^ source.ahk keyword.operator.arithmetic.ahk
# ^ source.ahk keyword.operator.arithmetic.ahk
#  ^ source.ahk keyword.operator.arithmetic.ahk
#   ^ source.ahk keyword.operator.arithmetic.ahk
#    ^ source.ahk keyword.operator.arithmetic.ahk
#     ^ source.ahk keyword.operator.arithmetic.ahk
#      ^ source.ahk keyword.operator.arithmetic.ahk
#       ^ source.ahk keyword.operator.arithmetic.ahk
#        ^ source.ahk keyword.operator.arithmetic.ahk
#         ^ source.ahk keyword.operator.arithmetic.ahk
#          ^ source.ahk keyword.operator.arithmetic.ahk
#           ^ source.ahk keyword.operator.arithmetic.ahk
#            ^ source.ahk keyword.operator.arithmetic.ahk
#             ^ source.ahk keyword.operator.arithmetic.ahk
#              ^ source.ahk keyword.operator.arithmetic.ahk
#               ^ source.ahk keyword.operator.arithmetic.ahk
#                ^ source.ahk keyword.operator.arithmetic.ahk
#                 ^ source.ahk keyword.operator.arithmetic.ahk
#                  ^ source.ahk keyword.operator.arithmetic.ahk
#                   ^ source.ahk keyword.operator.arithmetic.ahk
#                    ^ source.ahk keyword.operator.arithmetic.ahk
#                     ^ source.ahk keyword.operator.arithmetic.ahk
#                      ^ source.ahk keyword.operator.arithmetic.ahk
#                       ^ source.ahk keyword.operator.arithmetic.ahk
#                        ^ source.ahk keyword.operator.arithmetic.ahk
#                         ^ source.ahk keyword.operator.arithmetic.ahk
#                          ^ source.ahk keyword.operator.arithmetic.ahk
#                           ^ source.ahk keyword.operator.arithmetic.ahk
#                            ^ source.ahk keyword.operator.arithmetic.ahk
#                             ^ source.ahk keyword.operator.arithmetic.ahk
#                              ^ source.ahk keyword.operator.arithmetic.ahk
#                               ^ source.ahk keyword.operator.arithmetic.ahk
#                                ^ source.ahk keyword.operator.arithmetic.ahk
#                                 ^ source.ahk keyword.operator.arithmetic.ahk
#                                  ^ source.ahk keyword.operator.arithmetic.ahk
#                                   ^ source.ahk keyword.operator.arithmetic.ahk
#                                    ^ source.ahk keyword.operator.arithmetic.ahk
#                                     ^ source.ahk keyword.operator.arithmetic.ahk
#                                      ^ source.ahk keyword.operator.arithmetic.ahk
#                                       ^ source.ahk keyword.operator.arithmetic.ahk
#                                        ^ source.ahk keyword.operator.arithmetic.ahk
#                                         ^ source.ahk keyword.operator.arithmetic.ahk
#                                          ^ source.ahk keyword.operator.arithmetic.ahk
#                                           ^ source.ahk keyword.operator.arithmetic.ahk
#                                            ^ source.ahk keyword.operator.arithmetic.ahk
#                                             ^ source.ahk keyword.operator.arithmetic.ahk
#                                              ^ source.ahk keyword.operator.arithmetic.ahk
#                                               ^ source.ahk keyword.operator.arithmetic.ahk
#                                                ^ source.ahk keyword.operator.arithmetic.ahk
#                                                 ^ source.ahk keyword.operator.arithmetic.ahk
#                                                  ^ source.ahk keyword.operator.arithmetic.ahk
#                                                   ^ source.ahk keyword.operator.arithmetic.ahk
#                                                    ^ source.ahk keyword.operator.arithmetic.ahk
#                                                     ^ source.ahk keyword.operator.arithmetic.ahk
#                                                      ^ source.ahk keyword.operator.arithmetic.ahk
#                                                       ^ source.ahk keyword.operator.arithmetic.ahk
#                                                        ^ source.ahk keyword.operator.arithmetic.ahk
#                                                         ^ source.ahk keyword.operator.arithmetic.ahk
#                                                          ^ source.ahk keyword.operator.arithmetic.ahk
#                                                           ^ source.ahk keyword.operator.arithmetic.ahk
#                                                            ^ source.ahk keyword.operator.arithmetic.ahk
#                                                             ^ source.ahk keyword.operator.arithmetic.ahk
#                                                              ^ source.ahk keyword.operator.arithmetic.ahk
#                                                               ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                 ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                  ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                   ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                    ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                     ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                      ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                       ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                        ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                         ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                          ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                           ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                            ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                             ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                              ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                               ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                 ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                  ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                   ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                    ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                     ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                      ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                       ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                        ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                         ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                          ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                           ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                            ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                             ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                              ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                               ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                                ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                                 ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                                  ^ source.ahk keyword.operator.arithmetic.ahk
#                                                                                                   ^ source.ahk keyword.operator.arithmetic.ahk
>test2 := "
#^^^^^^ source.ahk
#      ^^ source.ahk keyword.operator.assignment.ahk
#        ^ source.ahk
#         ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
>        (  ;not has LTrim, this line deep === test2 line deep +1
#^^^^^^^^^ source.ahk string.multiline.ahk
#         ^ source.ahk string.multiline.ahk keyword.multiline.ahk
#          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk comment.line.semicolon.ahk
>deepRaw...
#^^^^^^^^^^^ source.ahk string.multiline.ahk
>    deepRaw...
#^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>            deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>deepRaw...
#^^^^^^^^^^^ source.ahk string.multiline.ahk
>                            deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>                        deepRaw...
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        )" ;this line deep === test2 line deep +1
#^^^^^^^^^^ source.ahk string.multiline.ahk
#          ^ source.ahk meta.comment.line.semicolon.ahk
#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>
>
>bar(str1, opt) {
#^^^ source.ahk entity.name.function.ahk
#   ^ source.ahk punctuation.bracket.ahk
#    ^^^^ source.ahk
#        ^ source.ahk punctuation.ahk
#         ^^^^ source.ahk
#             ^ source.ahk punctuation.bracket.ahk
#              ^ source.ahk
#               ^ source.ahk punctuation.bracket.ahk
>
>}
#^ source.ahk punctuation.bracket.ahk
>
>foo() {
#^^^ source.ahk entity.name.function.ahk
#   ^ source.ahk punctuation.bracket.ahk
#    ^ source.ahk punctuation.bracket.ahk
#     ^ source.ahk
#      ^ source.ahk punctuation.bracket.ahk
>
>    bar("
#^^^^ source.ahk
#    ^^^ source.ahk entity.name.function.ahk
#       ^ source.ahk punctuation.bracket.ahk
#        ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
>        (
#^^^^^^^^^ source.ahk string.multiline.ahk
>            w            -> up
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>            s            -> down
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>            a            -> left
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>            d            -> right
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.ahk string.multiline.ahk
>        )", "w",0)
#^^^^^^^^^^ source.ahk string.multiline.ahk
#          ^ source.ahk punctuation.ahk
#           ^ source.ahk
#            ^ source.ahk string.quoted.double.ahk punctuation.definition.string.begin.ahk
#             ^ source.ahk string.quoted.double.ahk
#              ^ source.ahk string.quoted.double.ahk punctuation.definition.string.end.ahk
#               ^ source.ahk punctuation.ahk
#                ^ source.ahk constant.numeric.decimal.ahk
#                 ^ source.ahk punctuation.bracket.ahk
>    ;    ^mulit-line end
#^^^ source.ahk
#   ^ source.ahk meta.comment.line.semicolon.ahk
#    ^^^^^^^^^^^^^^^^^^^^ source.ahk meta.comment.line.semicolon.ahk comment.line.semicolon.ahk
>}
#^ source.ahk punctuation.bracket.ahk
>